src/macros.rs

```diff
@@ -1,47 +1,250 @@
-/// Construct a BSON value
+// BSON macro based on the serde_json json! implementation.
+
+/// Construct a bson::BSON value from a literal.
+///
+/// ```rust
+/// # #[macro_use]
+/// # extern crate bson;
+/// #
+/// # fn main() {
+/// let value = bson!({
+///     "code": 200,
```

Comment:
Do we want to document the "=>" syntax, or are we considering that discouraged (but kept around for compatibility)?

Reply:
Considering it discouraged in favor of consistency, but here for backwards-compatibility.

Reply:
Sounds good